neqo-transport/src/recovery.rs

```diff
@@ -235,83 +252,108 @@ impl LossRecovery {
         ack_delay: Duration,
         now: Instant,
     ) -> (Vec<SentPacket>, Vec<SentPacket>) {
-        qdebug!([self] "ack received - largest_acked={}.", largest_acked);
-
-        let new_largest = self.space_mut(pn_space).update_largest_acked(largest_acked);
-        // If the largest acknowledged is newly acked and
-        // ack-eliciting, update the RTT.
-        if let Some(new_largest) = new_largest {
-            if new_largest.ack_eliciting {
-                let latest_rtt = now - new_largest.time_sent;
+        qdebug!([self] "ack received for {:?} - largest_acked={}.",
+                pn_space, largest_acked);
+
+        let (acked_packets, any_ack_eliciting) =
+            self.spaces.get_mut(pn_space).remove_acked(acked_ranges);
+        if acked_packets.is_empty() {
+            // No new information.
+            return (Vec::new(), Vec::new());
+        }
+
+        // Track largest PN acked per space
+        let space = self.spaces.get_mut(pn_space);
+        let largest_acked = max(space.largest_acked, Some(largest_acked)).expect("must be some");
```

Comment:
You use max() here and then != on the next line.  That's doubling up on checks.  You can just use `if Some(largest_acked) > space.largest_acked`.